framework/src/engine/generator/network_endpoint.ts

```diff
@@ -228,10 +228,10 @@ export class NetworkEndpoint extends BaseNetworkEndpoint {
 			try {
 				for (const transactionBytes of transactionsData.transactions) {
 					const transaction = Transaction.fromBytes(transactionBytes);
+					transaction.validate();
```

Comment:
if this fails, the error will not be `InvalidTransactionError`, so i think it will not apply penalty.. on the other hand, it will apply penalty from `_receiveTransaction`?